### DISCUSSION TITLE
Not being able to install dbeaver from sources.

### DISCUSSION THREAD

**Discussion by Rakib-mbstu** [dbeaver/dbeaver#34728] [DBeaver Development] [answered]
hi,
I am trying to build dbeaver from source. I could not able to install it despite following the tutorials. My jdk version is 17.0.1 and maven is 3.6.0. The error it shows are added below. Any help would greatly be appreciated.
![Screenshot from 2024-07-09 22-05-09](https://github.com/dbeaver/dbeaver/assets/48640183/7630eb96-3b5b-40d6-a60c-76ef49add8e0)

**Comment by Destrolaric** [comment, marked as answer]
Call
`mvn clean install` install on `dbeaver-common` before the launch, we will improve the scripts to do that automatically a bit later

**Reply by Rakib-mbstu** [reply]
now it's stuck on this
![Screenshot from 2024-07-10 18-06-54](https://github.com/dbeaver/dbeaver/assets/48640183/198b9860-435e-4b2f-9880-a1dd229a9917)

**Reply by Destrolaric** [reply]
Everything should be good if you have a proper internet connection. The generator indexes p2 repositories and downloads dependencies by itself. Do you have some kind of the firewall?

**Reply by Rakib-mbstu** [reply]
Don't have a firewall. It downloads alright in other commands from p2 repositories.

**Reply by Destrolaric** [reply]
What happens if you directly open this repository?

**Reply by Rakib-mbstu** [reply]
Now i am seeing another problem. I ran mvn clean install in org.jkiss.dbeaver.model.lsm and seeing `org.eclipse.tycho.DependencyResolutionException: Cannot resolve dependencies of MavenProject: org.jkiss.dbeaver:org.jkiss.dbeaver.model.lsm:1.0.30-SNAPSHOT @ /home/mir/IdeaProjects/dbeaver-os/dbeaver/plugins/org.jkiss.dbeaver.model.lsm/pom.xml
`
here is a screemshot
![Screenshot from 2024-07-13 13-51-37](https://github.com/user-attachments/assets/2fbfde54-5008-46e9-a62d-94f4c5b23fa0)

**Comment by Rakib-mbstu** [comment]
Well now it's working perfectly. I think the how to run on IDEA should be re-written. The steps aren't clear. Someone new to the project might have hard time building and running the project.

**Comment by Destrolaric** [comment]
I just committed some changes to the steps. They should appear after a week or two. Thanks for the feedback about the documentation.
